api/v1alpha1/etcdcluster_types_test.go

```diff
@@ -0,0 +1,22 @@
+package v1alpha1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+)
+
+var _ = Context("CalculateQuorumSize", func() {
+	It("should return correct result for odd number of replicas", func() {
```

Comment:
It also good idea to test with nil value in replicas

Reply:
Nil value is impossible due to kubebuilder marker
```go
// +kubebuilder:default:=3
```

Reply:
Then it's strange to have this value as a pointer in struct. If it always prefilled. But I guess it's out of scope of this PR. We can discuss it later